### PR TITLE
Explicitly mark home directories as having a posix or windows flavor

### LINKED ISSUE #891: Explicit path comparison for user home directories
Ensure that the paths to the home directory on both Unix and Windows are a `posix_path` and `windows_path`, respectively. This could cause problems, such as manifested in #866.
 
Also, adjust the unit test for the home directory on OSX  by removing the explicit type check on `posix_path`.

### PR DESCRIPTION
**Pull request by twiggler**
- Explicitly mark home directories as either `posix_path` or `windows_path`
- Add missing unit test for parsing of users on windows and unix

Closes #891

### CODE REVIEW

File: dissect/target/plugins/os/unix/_os.py [hunk 1]
```diff
@@ -6,6 +6,8 @@
 from struct import unpack
 from typing import Iterator, Optional, Union
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/_os.py:7]: Could you update type hints in this file to modern variations using `|`?

File: dissect/target/plugins/os/windows/_os.py [hunk 2]
```diff
@@ -4,6 +4,8 @@
 import struct
 from typing import Any, Iterator, Optional
```
Comment (Schamper) [inline dissect/target/plugins/os/windows/_os.py:5]: Could you update type hints in this file to modern variations using `|`?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: tests/plugins/os/unix/bsd/citrix/test__os.py [hunk 3]
```diff
@@ -51,15 +53,15 @@ def test_citrix_os(target_citrix: Target, fs_bsd: VirtualFilesystem) -> None:
...
 
     assert users[6].name == "root"  # User entry for /root, from the config
-    assert users[6].home == "/root"
+    assert users[6].home == posix_path("/root")
     assert users[6].shell is None
 
     assert users[7].name == "root"  # User entry for /root, from /etc/passwd
-    assert users[7].home == "/root"
+    assert users[7].home == posix_path("/root")
     assert users[7].shell == "/usr/bin/bash"
```
Comment (Schamper) [inline tests/plugins/os/unix/bsd/citrix/test__os.py:67]: Misc comment that may needs to be it's own PR, but I think `shell` should also be a proper path.

File: dissect/target/plugins/os/unix/_os.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 import re
 import uuid
 from struct import unpack
-from typing import Iterator, Optional, Union
+from typing import Iterator, Union
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/_os.py:7]: Can you also replace `Union`?

**Review by Schamper — CHANGES_REQUESTED** [review]
Only small question, looks good otherwise.

File: tests/plugins/os/unix/bsd/citrix/test__os.py (hunk 3 above)
Comment (twiggler) [reply tests/plugins/os/unix/bsd/citrix/test__os.py:67]: Yeah, probably.

It is marked as a string in the record definition. 
Because it is a string, at least it won´t suffer from the issue we had where the path got converted to the flavor of the  host OS. 
But it is semantically clearer to mark it as a path.

Will think about it some more and create a ticket.

File: dissect/target/plugins/os/unix/_os.py (hunk 4 above)
Comment (twiggler) [reply dissect/target/plugins/os/unix/_os.py]: Replaced

**Review by Schamper — APPROVED** [review]
If you make the title slightly more descriptive this looks good to me!
